src/package.cpp

```diff
@@ -29,8 +31,82 @@ Value cpath = Nil();
 Value path = Nil();
 Table loaded;
 Table preload;
-Table searchers;
+Table searchers = *new Table(
+    {{1,
+      [](const CallContext& ctx) -> Value {
+          auto name = ctx.arguments().get(0);
+          return preload.get(name);
+      }},
+     {2, 2}});
```

Comment:
This is a memory leak. You can just wrap it in a unique_ptr and use the raw pointer from that everywhere. The unique_ptr will automatically clean up when the program exits.

Also all of these variables should be static so they don't conflict with others. Unless they are also defined in a header you should always make all global variables in cpp files static.

In this context static means that only the current file can access the variable.